examples/lambda/wcbv-reasonable/LScript.sml

```diff
@@ -0,0 +1,576 @@
+(* See:
+     Forster, Kunze and Roth,
+     "The Weak Call-by-Value 𝜆-Calculus Is Reasonable for Both Time and Space", POPL 2020
+   for inspiration
+*)
+open HolKernel Parse boolLib bossLib;
+open arithmeticTheory;
+open listTheory;
+open PrelimsTheory;
+open pure_dBTheory;
+
+val _ = new_theory "L";
+
+(* ------------------
+	      CBV LC
+   ------------------ *)
+
+Definition size:
+	size s =
+		case s of
+			| dV x => 1 + x
+			| dABS s => 1 + (size s)
+			| dAPP s t => 1 + size s + size t
+End
+
+(* --------------------------------
+	  Substitution and Closedness
+   -------------------------------- *)
+
+Definition subst:
```

Comment:
This is the `sub` function from `pure_dB` with the arguments in a different order.  If you don't want to edit all the occurrences, use an overload to make `subst` an alias: 
```
Overload subst = “\s k u. sub u k s”
```

Reply:
In progress, finishing soon.. Note that `sub` from `pure_dB` allows open terms, whilst `subst` assumes closed terms, their definitions are actually of different natures (different lifting methods). I'm adding assumptions for terms being closed in the correctness proofs for time and space models in order to prove these theorems using `sub`.

Reply:
Apologies, you're quite right.  If the terms are always closed, then you have that `pure_dB$sub` is the same as this `subst` (as you say), but are there uses in the "correctness proofs" where the closed-ness isn't assumed?  That sounds weird: the result of such a substitution with `subst` is not a reasonable term at all.

Reply:
The original Coq proof does have the closed term assumption for the final proof for the time cost model for the heap machine but didn't include this assumption in the lemmas used by this final proof nor in the other models (i.e. the time cost model for the substitution machine and the space cost models for the substitution and the heap machines).

Reply:
Finished fixing all the main correctness proofs for both cost models for both machines! Left out some lemmas that require closed term assumptions but are not used in any of the later proofs for now (they are inside LScript, related to bound and closed).

Reply:
Will look at these lemmas tomorrow.. shouldn't be too hard to fix, I hope.